### ISSUE TITLE
Unable to start activity ComponentInfo: Invalid property name: İd

### ISSUE THREAD

**Issue by ezramorse** [AppGyver/steroids#1076]
We've had 5 of these in the last 3 days on with steroids Android build 4.1 and 4.2. Included are the sections of the config, which appear to be fine. The program runs on hundreds/thousands of devices without an issue (so I don't think its a configuration problem), but here and there we get this report:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.sevencupsoftea.app/com.appgyver.runtime.standalone.MainActivity}: java.lang.RuntimeException: Invalid property name: İd
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2429)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2493)
	at android.app.ActivityThread.access$800(ActivityThread.java:166)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1283)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:136)
	at android.app.ActivityThread.main(ActivityThread.java:5584)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:515)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1268)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1084)
	at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.RuntimeException: Invalid property name: İd
	at com.appgyver.app.ApplicationConfigParser.setPropertyValue(ApplicationConfigParser.java:337)
	at com.appgyver.app.ApplicationConfigParser.copyBigInt(ApplicationConfigParser.java:305)
	at com.appgyver.app.ApplicationConfigParser.setAccessibleFields(ApplicationConfigParser.java:146)
	at com.appgyver.app.ApplicationConfigParser.loadApplicationJson(ApplicationConfigParser.java:136)
	at com.appgyver.app.ApplicationConfigParser.loadApplicationJsonFromSteroidsBaseDir(ApplicationConfigParser.java:131)
	at com.appgyver.app.ApplicationDescription.loadApplicationJsonFromAssets(ApplicationDescription.java:390)
	at com.appgyver.android.application.AGAndroidApplication.loadSteroidsApplication(AGAndroidApplication.java:74)
	at com.appgyver.standalone.StandaloneApplication.loadSteroidsApplication(StandaloneApplication.java:17)
	at com.appgyver.android.application.ApplicationActivity.loadSteroidsApplication(ApplicationActivity.java:187)
	at com.appgyver.android.application.ApplicationActivity.onCreate(ApplicationActivity.java:72)
	at com.appgyver.standalone.StandaloneApplicationActivity.onCreate(StandaloneApplicationActivity.java:14)
	at com.appgyver.runtime.standalone.MainActivity.onCreate(MainActivity.java:11)
	at android.app.Activity.performCreate(Activity.java:5447)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1094)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2393)
```

Here is our JSON config:
```
{
  "drawers": {
    "left": {
      "id": "left-drawer",
      "location": "xxx",
      "showOnAppLoad": false,
      "widthOfDrawerInPixels": 250
    },
    "right": {
      "id": "right-drawer",
      "location": "xxx",
      "showOnAppLoad": false
    }
  },
  "initialView": {
    "id": "loader",
    "location": "http://localhost/initial.html"
  },
  "tabs": [
    {
      "position": 0,
      "id": "feed",
      "title": "Feed",
      "target_url": "xxx"
    },
    {
      "position": 1,
      "id": "chat",
      "title": "Chat",
      "target_url": "xxx"
    },
    {
      "position": 2,
      "id": "path",
      "title": "Path",
      "target_url": "xxx"
    },
    {
      "position": 3,
      "id": "community",
      "title": "Community",
      "target_url": "xxx"
    },
    {
      "position": 4,
      "id": "me",
      "title": "Me",
      "target_url": "xxx"
    }
  ],
  "configuration": {
    "extra_response_headers": {
      "Access-Control-Allow-Origin": "*",
      "Access-Control-Allow-Headers": "Content-Type, X-Requested-With"
    }
  }
}
```

Happened on (Android 4.1 to 6.0):
```
Galaxy S3 Mini (golden)	1	20.0%
Galaxy Tab3 Lite 7.0 (goyavewifi)	1	20.0%
HTC One (m7)	1	20.0%
General Mobile 4G Dual (gm4g_sprout)	1	20.0%
Galaxy Note II (t03g)	1	20.0%
```

Could this be related to: http://community.appgyver.io/t/invalid-property-name-initialview/3276

(Note the funny I, or "İ", in both cases)

**Comment by pre** [comment]
Internally, there's this in the parser:

`propertyName = propertyName.substring(0, 1).toUpperCase() + propertyName.substring(1);`

Here `propertyName` is presumably `id`. For some reason `toUpperCase("i")` will return `İ` instead of `I`. 

Do you have any information about which specific devices have this problem? Do you know which locale they might be using?

I'll dig around.

**Comment by pre** [comment]
Yes, this is definitely a locale problem, I'll provide a fix soon.

**Comment by pre** [comment]
This is now fixed in Android client version 4.2.1

**Comment by ezramorse** [comment]
No more crashes so far! Thanks for the quick work on this
